ticker/ticker.go

```diff
@@ -100,20 +109,24 @@ func (t *T) Ticks() <-chan time.Time {
 //
 // NOTE: Part of the Ticker interface.
 func (t *T) Resume() {
+	t.tickerMtx.Lock()
 	if t.ticker == nil {
 		t.ticker = time.NewTicker(t.interval)
 	}
+	t.tickerMtx.Unlock()
 }
 
 // Pause suspends the underlying ticker, such that Ticks() stops signaling at
 // regular intervals.
 //
 // NOTE: Part of the Ticker interface.
 func (t *T) Pause() {
+	t.tickerMtx.RLock()
```

Comment:
Looks like this should acquire the write lock instead.

Reply:
That's what I thought too but my test case didn't trigger a race when acquiring a read lock.  I talked with @cfromknecht and he suggested moving all of this locking behavior to the router where the actual race occurs.  So I'll do that I think

Reply:
Though with the approach I mentioned above, I'm not sure of a clean way to acquire a read lock before calling `case <-r.statTicker.Ticks():` in `networkHandler`. Thoughts?

Reply:
i think we just need to remove the `Resume()` call  in the goroutine to another place. One option is to do it at the top of the loop, and resume if `stats.Empty()`

Reply:
Oh right! Done